invenio_rdm_records/services/components/access.py

```diff
@@ -19,13 +20,24 @@ class AccessComponent(ServiceComponent):
 
     def _populate_access_and_validate(self, identity, data, record, **kwargs):
         """Populate and validate the record's access field."""
+        errors = []
         if record is not None and "access" in data:
+            access = data.get("access")
+
+            # Explicit permission check for modifying record access.
+            # NOTE: This is very static/inflexible, but generalizing per-field
+            # management permissions is difficult.
+            new_record_access = (access or {}).get("record")
+            can_manage = self.service.check_permission(identity, "manage_record_access")
+            if record.access.protection.record != new_record_access and not can_manage:
```

Comment:
minor: I know nested if's are ugly but we can save some db queries if we do the check _only_ if the values are diff:

```suggestion
            if record.access.protection.record != new_record_access:
                can_manage = self.service.check_permission(identity, "manage_record_access")
                if can_manage:
                ....
```